io-engine/src/bdev/nexus/nexus_child.rs

```diff
@@ -1103,16 +1103,13 @@ impl<'c> NexusChild<'c> {
         info!("{self:?}: unplugging child...");
 
         let state = self.state();
-        let is_destroying = self.is_destroying();
 
-        // Only drop the device and the device descriptor if the child is being
-        // destroyed. For a hot remove event, keep the device and descriptor.
-        if is_destroying {
-            debug!("{self:?}: dropping block device");
-            self.device = None;
-        } else {
-            debug!("{self:?}: hot remove: keeping block device");
-        }
+        // Drop the device and the device descriptor if the child is being
+        // destroyed or hot removed. For hot remove, the DeviceRemoved event
+        // will send us here and keeping the device/descriptor isn't of
+        // any use.
+        debug!("{self:?}: dropping block device");
+        self.device = None;
```

Comment:
hmm if there's a hot removal this way, does this mean we don't go through the pause dance?
If so then we need to cater for that as well..

Reply:
yeah we don't do that pause here because child retire isn't done. There are more nuances involved here than that make me think we are better maintain status quo with this code. I'm inclined to park this change for now.

1. A way to simulate hot remove is doing `bdev destroy` on pool bdev rather than `pool destroy`.  In this case, both kind of child(nvmf and spdk bdev) behave similarly and are unplugged fine, without going via retire path and pause.
2. The `pool destroy` or `replica destroy` are to be rather considered as fault. nvmf child receives `AdminCommandCompletionFailure` for these, deems the child faulted, **retires it, and then triggers `DeviceRemoved` event.** For spdk bdev, we directly get `DeviceRemoved` in these cases without child having gone through retire.
3. Practically, 1 is more likely than 2 in actual system.